infra/utils/validator/service-worker-runtime.js

```diff
@@ -0,0 +1,57 @@
+const assert = require('assert');
+const expect = require('chai').expect;
+const fse = require('fs-extra');
+const makeServiceWorkerEnv = require('service-worker-mock');
+const sinon = require('sinon');
+const vm = require('vm');
+
+module.exports = async ({swFile, swCode, expectedMethodCalls}) => {
```

Comment:
This code will pretty much destroy the testing changes coming in because it'll override the SW env stuff. Not something actionable on this PR, but more an FYI.

Reply:
This code uses a handcrafted `context` object that's passed to  `vm.runInNewContext()` and I don't anticipate interference due to the `global` state from other tests.

The forthcoming `generate-sw.js` e2e test may make use of `NODE_ENV`, but it will be specifically set to the values expected prior to running the tests.

I can refactor the spy setup and validation into separate functions.